concourse/steps/component_descriptor.mako

```diff
@@ -76,15 +77,28 @@ component_v2 = base_descriptor_v2.component
 
 ## XXX unify w/ injection-method used for main-repository
 % for repository in other_repos:
+<%
+  repo_variable_name = f"repo_{other_repos.index(repository)}"
+%>
+${repo_variable_name}_repo_path = os.path.abspath('${repository.resource_name()}')
+${repo_variable_name}_git_repo = git.Repo(${repo_variable_name}_repo_path)
+if not ${repo_variable_name}_git_repo.head.is_valid():
+  ${repo_variable_name}_commit_hash = commit_hash
+else:
+  try:
+    ${repo_variable_name}_commit_hash = ${repo_variable_name}_git_repo.head.commit.hexsha
+  except:
+    ${repo_variable_name}_commit_hash = commit_hash
+
 component_v2.sources.append(
   cm.ComponentSource(
     name='${repository.name()}',
     type=cm.SourceType.GIT,
     access=cm.GithubAccess(
       type=cm.AccessType.GITHUB,
-      repoUrl='${repository.repo_hostname()}/${repository.repo_path()}',
+      repoUrl='${repository.repo_hostname() or default_github_config.hostname()}/${repository.repo_path()}',
       ref='${repository.branch()}',
-      commit=commit_hash,
+      commit=${repo_variable_name}_commit_hash,
```

Comment:
as indicated before: please do not do this. this is unnecessarily hard to understand/read